hazelcast/src/main/java/com/hazelcast/sql/SqlService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+/**
+ * Service to execute SQL queries.
+ */
+public interface SqlService {
+    /** Unique service name. */
+    String SERVICE_NAME = "hz:impl:sqlService";
+
+    /**
+     * Execute query.
+     *
+     * @param sql SQL.
+     * @param params Parameters.
+     * @return Cursor.
+     */
+    default SqlResult query(String sql, Object... params) {
```

Comment:
in Jet we have `@Nonnull` and `@Nullable` annotations for all public method return types and parameters, I think this is done in IMDG as well, seems to be missing here perhaps?

Reply:
Added annotations to public API. Please double-check that it looks good now.